Added Auth0.Result type info (#8650)
auth0/docs: articles/libraries/auth0-swift/index.md

The first step in adding authentication to your iOS application is to provide a way for your users to log in. The fastest, most secure, and most feature-rich way to do this with Auth0 is to use <dfn data-key="universal-login">Universal Login</dfn>.

::: note

> If you need help between the two types of login flows, refer to [Browser-Based vs. Native Login Flows on Mobile Devices](/tutorials/browser-based-vs-native-experience-on-mobile)

Reviewer: I would reword the start of this a bit. I suggest:

```suggestion
For more information on the two types of login flows, please refer to [Browser-Based vs. Native Login Flows on Mobile Devices](/tutorials/browser-based-vs-native-experience-on-mobile)
```

    }
```

> If you're using **Swift 5+** the `Auth0.Result` type may shadow Swift's `Result`. To prevent that, whenever you want to refer to Swift's type, use `Swift.Result` instead. This will be fixed in the next major version of Auth0.swift.

Reviewer: ```suggestion
If you're using **Swift 5+**, `Auth0.Result` may shadow Swift's `Result` type. To prevent that, replace it with `Swift.Result` whenever you want to refer to Swift's built-in type. This will be fixed in the next major version of Auth0.swift.
```

Author: Thanks @stevehobbsdev!
